mlir/lib/Dialect/Linalg/IR/LinalgOps.cpp

```diff
@@ -1217,9 +1217,8 @@ struct EraseIdentityLinalgOp : public OpRewritePattern<OpTy> {
 
   LogicalResult matchAndRewrite(OpTy linalgOp,
                                 PatternRewriter &rewriter) const override {
-    // Check all indexing maps are identity.
-    if (llvm::any_of(linalgOp.getIndexingMapsArray(),
-                     [](AffineMap map) { return !map.isIdentity(); }))
+    // All indexing maps must be equal. It follows that they are permutations.
+    if (!llvm::all_equal(linalgOp.getIndexingMapsArray()))
```

Comment:
Do we also check if they are permutation? I'm a bit worried about something like `#map = affine_map<(d0, d1, d2) -> (d0, d1 + d2)>` or something with symbols (e.g., `s0`).

Reply:
It is hard to provide an example now, but people could generate uncommon linalg ops.

Reply:
oh, I missed the comment from Mahesh. After reading more codes and IR examples, perhaps it is not needed. Because they are invalid linalg ops with these affine maps.

Reply:
I was worried about indexing maps like conv, and I realized that we cant have every maps being complicated maps.

https://github.com/llvm/llvm-project/blob/bc747c3e1377e3e262d901cefcae2f563c895bb5/mlir/test/Dialect/Linalg/generalize-named-ops.mlir#L136-L138

Reply:
Rest of the logic should cover the "copy" semantics, but if the input and output have same indexing map, then it shouldnt really matter.